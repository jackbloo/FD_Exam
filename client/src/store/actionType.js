export const LOGIN = 'LOGIN'
export const FETCHPASSWORD = 'FETCHPASSWORD'
export const LOGOUT = 'LOGOUT'
export const FETCHDETAIL = 'FETCH DETAIL'
export const SETEDIT = 'SETEDIT'
export const SHOW = 'SHOW'
export const UNSHOW = 'UNSHOW'
export const LOAD = 'LOAD'
export const UNLOAD = 'UNLOAD'
export const DELETE = 'DELETE'