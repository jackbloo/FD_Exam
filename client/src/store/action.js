import {LOGIN, LOGOUT, FETCHPASSWORD,SHOW,UNSHOW,LOAD,UNLOAD, FETCHDETAIL} from '../store/actionType'


export const show = () => ({type:SHOW})
export const unshow = () => ({type:UNSHOW})
export const login = () => ({type:LOGIN})
export const logout = () => ({type:LOGOUT})
export const load = () => ({type:LOAD})
export const unload = () => ({type: UNLOAD})
export const fetchPassword = (data) =>  ({type:FETCHPASSWORD, data})


// export const deleteOnePass = (id) => dispatch => {
//     dispatch({type:LOAD})
//     pm
//     .doc(id)
//     .delete()
//     .then(function() {
//         dispatch({type:UNLOAD})
//         toast.success('Your Password is deleted')
//     }).catch(function(error) {
//         toast.error(error)
//     });
// }
