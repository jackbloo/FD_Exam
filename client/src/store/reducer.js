import {GETDATA} from './actionType'

const initialState = {
    editorChoice:[],
    latestArticles:[],
    latestReview:[],
    navbar: [{id:'skincare'},{id:'make up'},{id:'body'},{id:'hair'},{id:'fragrance'},{id:'nails'},{id:'tools'},{id:'brands'}],
    linkData:[{id: 'About Us'}, {id: 'Terms & Conditions'}, {id: 'Awards'},{id: 'Feedback'},{id: 'Privacy Policy'}, {id: 'Newsletter'}, {id: 'Content'},{id: 'Help'}]
}

const reducer = (state = initialState, action) => {
    switch(action.type){
        case GETDATA:
            const data = action.data
            return {
                ...state,
                editorChoice: [...data["editor's choice"]],
                latestArticles:[...data["latest articles"]],
                latestReview:[...data["latest review"]]
            }
        default:
            return state     
    }
 
}

export default reducer