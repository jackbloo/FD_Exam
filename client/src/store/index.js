import { configureStore } from '@reduxjs/toolkit'
import reducer from '../store/reducer'

const store = configureStore({reducer: reducer})

export default store