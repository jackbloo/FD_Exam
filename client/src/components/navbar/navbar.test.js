import React from 'react';
import {render, screen} from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Navbar from './navbar';
import { Provider} from 'react-redux'
import store from '../../store'


describe('Unit Testng of Navbar', () => {
    it('Should Render Navbar match snapshot', ()=>{

     render(
            <Provider store={store}>
                    <Navbar/>
            </Provider>
        )
        const card = screen.getByTestId('navbar-container')
        expect(card).toMatchSnapshot()
    })


    
    it('Should Render Navbar without crashing', ()=>{
     render(
            <Provider store={store}>
                    <Navbar/>
            </Provider>
        )
        const card = screen.getByTestId('navbar-container')
        expect(card).toBeInTheDocument()
        expect(card).toBeVisible()
        expect(card).toHaveTextContent("LOGIN/SIGNUPSKINCAREMAKE UPBODYHAIRFRAGRANCENAILSTOOLSBRANDS")
        expect(card).not.toHaveTextContent('Facebook.com')
    })

    
    it('Should Render navbar burgerbar', ()=>{
     render(
            <Provider store={store}>
                 <Navbar/>
            </Provider>
        )
        const card = screen.getByTestId('navbar-burgerbar')
        expect(card).toBeInTheDocument()
        expect(card).toBeVisible()
    })

    it('Should Render navbar Female Daily logo', ()=>{
        render(
               <Provider store={store}>
                      <Navbar/>
               </Provider>
           )
           const card = screen.getByTestId('navbar-fdlogo')
           expect(card).toBeInTheDocument()
           expect(card).toBeVisible()
       })

       it('Should Render navbar Search icon', ()=>{
        render(
               <Provider store={store}>
                      <Navbar/>
               </Provider>
           )
           const card = screen.getByTestId('navbar-search-icon')
           expect(card).toBeInTheDocument()
           expect(card).toBeVisible()
       })

       it('Should Render navbar Search Bar', ()=>{
        render(
               <Provider store={store}>
                 <Navbar/>
               </Provider>
           )
           const card = screen.getByTestId('navbar-searchbar')
           expect(card).toBeInTheDocument()
           expect(card).toBeVisible()
       })

       it('Should Render navbar login and signup', ()=>{
        render(
               <Provider store={store}>
               <Navbar/>
               </Provider>
           )
           const card = screen.getByTestId('navbar-login-signup')
           expect(card).toBeInTheDocument()
           expect(card).toBeVisible()
       })


})

