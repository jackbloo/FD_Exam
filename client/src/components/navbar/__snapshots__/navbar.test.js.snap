// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit Testng of Navbar Should Render Navbar match snapshot 1`] = `
<div
  class="sc-bdvvtL eXDUeH"
  data-testid="navbar-container"
>
  <div
    class="sc-gsDKAQ fEybuc"
  >
    <div
      class="sc-hKwDye jpPtVJ"
    >
      <img
        class="sc-furwcr IVlna"
        data-testid="navbar-burgerbar"
        src="/assets/burger-bar.png"
      />
      <img
        class="sc-iCfMLu dWTIif"
        data-testid="navbar-fdlogo"
        src="/assets/logo.png"
      />
    </div>
    <div
      class="sc-eCImPb fGQdog"
    >
      <i
        class="sc-jrQzAO cZEjhg fa fa-search icon"
        data-testid="navbar-search-icon"
      />
      <input
        class="sc-pVTFL hoXjtW"
        data-testid="navbar-searchbar"
        placeholder="Search products, articles, topics, brands, etc"
      />
    </div>
    <div
      class="sc-jRQBWg jpWkkf"
    >
      <i
        class="sc-kDTinF cpjqwz fa fa-user icon"
      />
      <div
        class="sc-iqseJM dcWleQ"
        data-testid="navbar-login-signup"
      >
        LOGIN/SIGNUP
      </div>
    </div>
  </div>
  <div
    class="sc-dkPtRN gmTEFR"
  >
    <div
      class="sc-gKclnd kIFiRD"
    >
      SKINCARE
    </div>
    <div
      class="sc-gKclnd kIFiRD"
    >
      MAKE UP
    </div>
    <div
      class="sc-gKclnd kIFiRD"
    >
      BODY
    </div>
    <div
      class="sc-gKclnd kIFiRD"
    >
      HAIR
    </div>
    <div
      class="sc-gKclnd kIFiRD"
    >
      FRAGRANCE
    </div>
    <div
      class="sc-gKclnd kIFiRD"
    >
      NAILS
    </div>
    <div
      class="sc-gKclnd kIFiRD"
    >
      TOOLS
    </div>
    <div
      class="sc-gKclnd kIFiRD"
    >
      BRANDS
    </div>
  </div>
</div>
`;
